ecosystem/gstreamer_plugin/gst_mtl_st20p_tx.c

```diff
@@ -304,14 +306,19 @@ static gboolean gst_mtl_st20p_tx_session_create(Gst_Mtl_St20p_Tx* sink, GstCaps*
     return FALSE;
   }
 
-  if (sink->framerate) {
-    if (!gst_mtl_common_parse_fps_code(sink->framerate, &ops_tx.fps)) {
-      GST_ERROR("Failed to parse custom ops_tx fps code %d", sink->framerate);
+  if (sink->fps_n > 0 && sink->fps_d > 0) {
+    ops_tx.fps = st_frame_rate_to_st_fps((double)sink->fps_n / sink->fps_d);
+    if (ops_tx.fps == ST_FPS_MAX) {
+      GST_ERROR("Invalid framerate: %d/%d", sink->fps_n, sink->fps_d);
+      return FALSE;
+    }
+  } else {
+    ops_tx.fps = st_frame_rate_to_st_fps((double)info->fps_n / info->fps_d);
```

Comment:
are we sure the info->fps_d will never be 0 ?

Reply:
I check that, GST_TYPE_FRACTION is always a valid fraction, I was unable to initialize it to 1/0. 

I added a initialization to default values for all fps variables so it won't be 0 if unset.

Reply:
now, I'm thinking that my assumption that info->fps_d is always set by GST_TYPE_FRACTION could be incorrect. It is like that if for example rawvideoparse is used, but it actually could be 0. I will add a check.

Reply:
lovly